taichi/ir/frontend_ir.cpp

```diff
@@ -586,6 +590,9 @@ void LoopUniqueExpression::flatten(FlattenContext *ctx) {
 
 void IdExpression::flatten(FlattenContext *ctx) {
   stmt = ctx->current_block->lookup_var(id);
+  if (!ret_type->is_primitive(PrimitiveTypeID::unknown)) {
+    stmt->ret_type = ret_type;
+  }
```

Comment:
Is this necessary? It seems that it has already been covered by 
https://github.com/taichi-dev/taichi/blob/5774ea62166456a9ec6ce4f69534527ba437bfe8/taichi/ir/frontend_ir.cpp#L27-L33

Reply:
This is a workaround for type checking a FrontendAssignStmt. Since the type info only propagates in the constructor, we need its type to propagate the type to the variable being assigned.